snuba/web/constants.py

```diff
@@ -0,0 +1,33 @@
+from clickhouse_driver.errors import ErrorCodes
+
+from snuba.clickhouse.errors import ClickhouseError
+
+### HTTP Status Codes ###
+INTERNAL_SERVER_ERROR = 500
+BAD_REQUEST = 400
+
+### ClickHouse Error Codes ###
+
+# Since the query validator doesn't have a typing system, queries containing type errors are run on
+# Clickhouse and generate ClickhouseErrors. Return a 400 status code for such requests because the problem
+# lies with the query, not Snuba.
+CLICKHOUSE_TYPING_ERROR_CODES = {
+    ErrorCodes.ILLEGAL_TYPE_OF_ARGUMENT,
+    ErrorCodes.TYPE_MISMATCH,
+}
```

Comment:
Seems the original change was introduced here: https://github.com/getsentry/snuba/pull/2672

Reply:
@fpacifici yes you are right, and please see the PR Rahul linked for the context. Our rationale for doing it this way is:
* Detecting type problems in the validator requires adding a type system to the validator, which would be a lot of work. It's quite possible that such a type system would itself have bugs, and cause the sort of problem you are raising.
* If a bug of the sort you described is introduced, then we'd see errors in the Sentry project so we'd find out about those there. As things stood prior to this change, if Sentry sent us an incorrectly typed query, it would cause errors in both Sentry and Snuba which is arguably redundant.

Reply:
Sure, my question was for Evan/Volo

Reply:
Sure, sounds good.